src/lib.rs

```diff
@@ -499,11 +499,36 @@
 //! #[derive(StructOpt)]
 //! struct Foo {
 //!   #[structopt(long = "secret", env = "SECRET_VALUE", hide_env_values = true)]
-//!   param: String
+//!   secret_value: String
+//! }
+//! ```
+//!
+//! ### Auto-deriving environment variables
+//!
+//! Environment variables tend to be called after the corresponding `struct`'s field,
+//! as in example above. The field is `secret_value` and the env var is "SECRET_VALUE";
+//! the name is the same, except casing is different.
+//!
+//! It's pretty tedious and error-prone to type the same name twice,
+//! so you can ask `structopt` to do that for you.
+//!
+//! ```
+//! # use structopt::StructOpt;
+//!
+//! #[derive(StructOpt)]
+//! struct Foo {
+//!   #[structopt(long = "secret", env)]
+//!   secret_value: String
 //! }
-//! # fn main() {}
 //! ```
 //!
+//! It works just like `#[structopt(short/long)]`: if `env` is not set to some concrete
+//! value the value will be derived from the field's name. This is controlled by
+//! `#[structopt(rename_all_env)]`.
+//!
+//! `rename_all_env` works almost exactly as `rename_all` (including overriding)
```

Comment:
```suggestion
//! `rename_all_env` works exactly as `rename_all` (including overriding)
```
Else we might think there is some subtle undocumented differences.